src/posthog-core.ts

```diff
@@ -1075,7 +1075,7 @@ export class PostHog {
         }
 
         const deviceIdMarksForIdentify = !this.get_property('$device_id')
-        const isKnownAnonymous = this.persistence.get_user_state() === 'anonymous'
+        const isKnownAnonymous = this.persistence.get_user_state() !== 'identified'
```

Comment:
This is not something that's obvious to understand -  could we explain why we're doing this, see also https://github.com/PostHog/posthog-js/issues/512#issuecomment-1421185109